packages/components/src/tab-panel/types.ts

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import type { ReactNode } from 'react';
+
+type Tab = {
+	/**
+	 * The key of the tab.
+	 */
+	name: string;
+	/**
+	 * The label of the tab.
+	 */
+	title: string;
+	/**
+	 * The class name to apply to the tab button.
+	 */
+	className?: string;
+} & Record< any, any >;
+
+export type TabButtonProps = {
+	children: ReactNode;
+	className?: string;
+	onClick: ( event: MouseEvent ) => void;
+	selected: boolean;
+	tabId: string;
+};
+
+export type TabPanelProps = {
+	/**
+	 * The class name to add to the active tab.
+	 *
+	 * @default 'is-active'
+	 */
+	activeClass?: string;
+	/**
+	 * A function which renders the tabviews given the selected tab.
+	 * The function is passed the active tab object as an argument as defined by the tabs prop.
+	 */
+	children: ( tab: Tab ) => ReactNode;
+	className?: string;
```

Comment:
Maybe add an description to the classname as well.

```suggestion
	/**
	 * The class name to give to the outer container for the TabPanel
	 */
	className?: string;
```

Reply:
Ah yes, thanks!